sktime/forecasting/rbf_forecaster.py

```diff
@@ -0,0 +1,575 @@
+        self.centers = centers
+        self.gamma = gamma
+        self.rbf_type = rbf_type
+        self.linear_layers = linear_layers
+        self.optimizer = optimizer
+        self.lr = lr
+        self.epochs = epochs
+        self.stride = stride
+        self.criterion = criterion
+
+        self.use_cuda = use_cuda
+
+        if self._check_torch_availability():
+            import torch
+
+        self._cuda_available = torch.cuda.is_available()
+        self._use_cuda_actual = self.use_cuda and self._cuda_available
+        self.device = torch.device("cuda" if self._use_cuda_actual else "cpu")
+
+        self.model = None
+        self.scaler = StandardScaler()
+
+    def _check_torch_availability(self):
+        """Check if torch is available and raise appropriate error if not."""
+        if not _check_soft_dependencies("torch", severity="none"):
+            raise ModuleNotFoundError(
+                "torch is a required optional dependency for RBFForecaster. "
+                "Please install torch to use this forecaster. "
+                "Instructions can be found at https://pytorch.org/get-started/locally/"
+            )
+        return True
+
+    def build_network(self, input_size):
+        """Build the RBF network architecture."""
+        output_size = 1
+
+        self.model = RBFNetwork(
+            input_size=input_size,
+            hidden_size=self.hidden_size,
+            output_size=output_size,
+            centers=self.centers,
+            gamma=self.gamma,
+            rbf_type=self.rbf_type,
+            linear_layers=self.linear_layers,
+        ).to(self.device)
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the RBF-based model to the provided time series data.
+
+        Parameters
+        ----------
+        y : pd.Series
+            The time series data to be fitted.
+        X : optional
+            Additional exogenous data (not used).
+        fh : optional
+            Forecasting horizon (not used).
+        """
+        import torch
+        from torch.utils.data import DataLoader, TensorDataset
+
+        self._y = y.copy()
+        if isinstance(y, pd.Series):
+            y_scaled = self.scaler.fit_transform(y.values.reshape(-1, 1))
+        else:
+            y_scaled = self.scaler.fit_transform(y.values)
+
+        X_train, y_train = self._create_windows(y_scaled)
+        X_tensor = torch.FloatTensor(X_train).to(self.device)
+        y_tensor = torch.FloatTensor(y_train).to(self.device)
+
+        dataset = TensorDataset(X_tensor, y_tensor)
+        dataloader = DataLoader(dataset, batch_size=self.batch_size, shuffle=True)
+
+        input_size = self.window_length
+        self.build_network(input_size)
+
+        criterion = self._instantiate_criterion()
+        optimizer = None
+
+        self.model.train()
+        for epoch in range(self.epochs):
+            total_loss = 0
+            for batch_X, batch_y in dataloader:
+                if optimizer is None:
```

Comment:
I created the the layers using "lazy initialisation" that i read somewhere, as the linear layer shape is defined after the fitting of `rbf-layer`, so the parameters are not known at the initialisation. That's why I added it in the loop only. If you want me to remove the RBFLayer class then it might change (not sure it will, as then i will remove the lazylinear class as well and everything will be done in _fit or in __init__ only)

Reply:
I mean the initialisation of the optimiser. Is the needed in the for loop?

Reply:
Actually the the parameters required to initialise the optimizer are not present before, as the shapes and all of the layers depends on the output of the transformer and rn it is happening in the loop only, so although optimizer is initialised only once in the first epoch only after that due to if statement it is not executed, it is required in the loop only, but now as you are saying I should transform it only once before fitting, that would change the things ig... I will look into it